#167729775-Ch-Integrating-Travis-CI

davidfrank96 opened the pull request:
#### What does this PR do?
Integrated Travis CI for continuous integration for deploying with confidence 

#### Description of Task to be completed?
- implemented a travis.yml file for Travis cloud to detect

#### How should this be manually tested?
pull or fetch this branch  
run npm install to add necessary dependencies

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
ch-Integrating-Travis-ci-167729775
#### Screenshots (if appropriate)
![Travis](https://user-images.githubusercontent.com/43038045/63087874-08375b80-bf08-11e9-9edf-dda5cc00d192.JPG)

#### Questions:

davidfrank96 committed:
Chore(Travis): implement exactly once delivery
- ensure i used the travis templates env, services,before_script and after success
- i implemented the travis.yml file
[Delivers #167729775]

davidfrank96 committed:
chore(Travis): implement exactly once delivery
- Updated the travis script
[Delivers #167729775]

davidfrank96 committed:
Chore(Travis): implement exactly once delivery
-changed the build and the test script for travis to pass
[Delivers #167729775]

davidfrank96 committed:
Chore(Travis): implement exactly once delivery
- removing database since this features has not been implemented yet
[Delivers #167729775]

davidfrank96 committed:
chore(Travis): implement exactly once delivery
[Delivers #167729775]

davidfrank96 committed:
Merge branch 'staging' into ch-Integrating-Travis-ci-167729775

davidfrank96 committed:
Merge branch 'staging' into ch-Integrating-Travis-ci-167729775

davidfrank96 committed:
chore(TravisCI)
implented the travis script
-postgres db
-sequelize
-Secret key
[Finishes #167729775]

davidfrank96 committed:
Merge branch 'staging' into ch-Integrating-Travis-ci-167729775

davidfrank96 committed:
chore(TravisBadges)
- included a markdown code for build badge on the READme.md file
[Finishes #167729775]

okabamac reviewed (approved):
Looks good to me and Travis is now working. Good job

chibytez reviewed (approved):
gone through it, looks great, please notify others of the port and secret key

Stylll commented on .travis.yml:
```diff
@@ -0,0 +1,32 @@
+   - TEST_DATABASE_URL=postgres://postgres:@127.0.0.1:5432/test_database
+   - SECRET_KEY=secretkey
+   - NODE_ENV=test
+   - SOCKET_PORT=3000
```
@davidfrank96 We should not be displaying our global variables like this. Travis provides a key/value pair store where these variables can be stored and then referenced here. Please look into that.

davidfrank96 replied:
thank you for the kind words.. will implement that ASAP

Stylll reviewed (changes requested):
Hey @davidfrank96 Please check out my comment.

Stylll commented on .travis.yml:
```diff
@@ -0,0 +1,26 @@
+before-script:
+ - npm add sequelize-cli
+
+before_script:
```
@davidfrank96 You have two before_script sections. Can you combine them into 1?

davidfrank96 replied:
Okay...

davidfrank96 committed:
chore(Secretkey)
- Removed the secretkey from the travis.yml
- Removed all the env files to Travis stationary
[Finishes #167729775]

davidfrank96 committed:
Chore(beforeScript)
- combined two before script to one.
[Finishes #167729775]
